src/st/cat/e_hellfire_beast.c

```diff
@@ -0,0 +1,1189 @@
+        prim = FindFirstUnkPrim(prim);
+        if (prim != NULL) {
+            prim->p3 = 1;
+            prim->p1 = 0;
+            prim->y0 = prim->y1 = self->posY.i.hi;
+            prim->y2 = prim->y3 = self->posY.i.hi;
+            prim->drawMode = DRAW_TPAGE2 | DRAW_TPAGE | DRAW_COLORS |
+                             DRAW_UNK02 | DRAW_TRANSP;
+        }
+        self->ext.hellfireBeastFlamePillar.unk80 = 0x18;
+        self->ext.hellfireBeastFlamePillar.castTimer = -0x10;
+        PlaySfxPositional(SFX_FIREBALL_SHOT_A);
+        self->step++;
+
+        break;
+    case 3:
+        prim = self->ext.hellfireBeast.primThree;
+        prim->drawMode = DRAW_DEFAULT;
+
+        prim = prim->next;
+        prim->drawMode = DRAW_UNK_800;
+
+        // NB. Strange way to write a single loop iteration...
+        // Maybe the number of iterations changed during development?
+        for (i = 0; i <= 0; i++) {
+            newEntity = AllocEntity(&g_Entities[224], &g_Entities[256]);
+            if (newEntity != NULL) {
+                // func_us_801CD614
+                CreateEntityFromCurrentEntity(E_UNK_35, newEntity);
+                newEntity->posX.i.hi = Random() & 0xF;
+                newEntity->posY.i.hi = 0x20;
+                newEntity->params = Random() & 1;
+                newEntity->ext.hellfireBeastThorsHammer.entity = self;
+            }
+        }
+
+        prim = self->ext.hellfireBeastFlamePillar.prim;
+        while (prim != NULL) {
+            if (prim->p3) {
+                func_us_801CD728(prim);
+            }
+            prim = prim->next;
+        }
+
+        prim = self->ext.hellfireBeastFlamePillar.prim;
+        while (prim != NULL) {
+            prim->p1 += 2;
+            prim = prim->next;
+        }
+
+        self->ext.hellfireBeastFlamePillar.castTimer++;
+        if (self->ext.hellfireBeastFlamePillar.castTimer < 0) {
+            prim = self->ext.hellfireBeastFlamePillar.prim;
+            while (prim != NULL) {
+                prim->p1 += 6;
+                prim = prim->next;
+            }
+
+            if (self->ext.hellfireBeastFlamePillar.unk80 < 0x20) {
+                self->ext.hellfireBeastFlamePillar.unk80++;
+            }
+        } else {
+            self->hitboxWidth = 0xC;
+            self->hitboxHeight = 0x30;
+            self->hitboxOffX = -4;
+            self->hitboxOffY = -0x30;
+            if (!(self->ext.hellfireBeastFlamePillar.castTimer % 8)) {
+                PlaySfxPositional(SFX_EXPLODE_B);
+            }
+        }
+        if (self->ext.hellfireBeastFlamePillar.castTimer > 0x60) {
+            self->ext.hellfireBeastFlamePillar.unk80--;
+            self->hitboxState = 0;
+            if (self->ext.hellfireBeastFlamePillar.unk80 < 0) {
+                DestroyEntity(self);
+            }
+        }
+        break;
+    }
+}
+
+extern EInit g_EInitHellfireBeastHitbox;
+void EntityHellfireBeastHitbox(Entity* self) {
```

Comment:
I'm not sure this is an accurate name for this.  Generally, these are the attack hitboxes that are independent from the normal hitbox.
I've been calling them EntityXAttack.